api/api/router.go

```diff
@@ -66,113 +109,35 @@ func (route Route) HandlerFunc(validate *validator.Validate) http.HandlerFunc {
 					return BadRequest(errMessage)
 				}
 			}
-			return route.handler(r, vars, body)
+			return route.Handler(r, vars, body)
 		}()
 
 		response.WriteTo(w)
 	}
 }
 
-func NewRouter(appCtx AppContext) *mux.Router {
+func NewRouter(appCtx *AppContext, controllers []Controller) *mux.Router {
```

Comment:
Thanks for refactoring this to keep this method lean by isolating the initialisation of controllers and their routes elsewhere.